inflation/lp/InflationLP.py

```diff
@@ -0,0 +1,1862 @@
+        op_string : str
+            String representation of an operator, e.g., ``"B_2_1_3_4"``.
+
+        Returns
+        -------
+        numpy.ndarray
+            2D array encoding of the operator.
+        """
+        components = op_string.replace('∅','0').split("_")
+        assert len(components) == self._nr_properties, \
+            f"There need to be {self._nr_properties} properties to match " + \
+            "the scenario."
+        components[0] = self.names_to_ints[components[0]]
+        return np.array([int(s) for s in components], dtype=self.np_dtype)
+
+    ###########################################################################
+    # ROUTINES RELATED TO THE GENERATION OF THE LP                            #
+    ###########################################################################
+
+    def build_raw_lexboolvecs(self) -> Tuple[np.ndarray, np.ndarray]:
+        r"""Creates the generating set of monomials (as boolvecs),
+        both in and out of Collins-Gisin notation.
+        """
+        choices_to_combine = []
+        lengths = []
+        if not self.nonfanout:
+            for raw_boolarray in self.all_ortho_groups_as_boolarrays:
+                boolvecs = np.pad(raw_boolarray, ((1, 0), (0, 0)))
+                lengths.append(len(boolvecs))
+                choices_to_combine.append(boolvecs)
+        else:
+            for party in range(self.nr_parties):
+                boolvecs = \
+                    self.InflationProblem._generate_compatible_monomials_given_party(
+                        party, with_last_outcome=True)
+                lengths.append(len(boolvecs))
+                choices_to_combine.append(boolvecs)
+        # Use reduce to take outer combinations, using bitwise addition
+        if self.verbose > 0:
+            eprint(f"About to generate {np.prod(lengths)} probability placeholders...")
+
+        choices_to_combine_CG = []
+        choices_to_combine_global = []
+        for choices in choices_to_combine:
+            CG_selection = np.logical_not(np.matmul(choices, 
+                                                    self.boolvec_for_CG_ineqs))
+            choices_to_combine_CG.append(choices[CG_selection])
+            event_count = choices.sum(axis=1)
+            choices_to_combine_global.append(choices[event_count == event_count.max()])
+        raw_boolvecs_CG = reduce(nb_outer_bitwise_or, 
+                                 reversed(choices_to_combine_CG))
+        raw_boolvecs_global = reduce(nb_outer_bitwise_or,
+                                     reversed(choices_to_combine_global))
+        return (raw_boolvecs_CG[np.argsort(raw_boolvecs_CG.sum(axis=1))],
+                raw_boolvecs_global[np.argsort(np.matmul(raw_boolvecs_global,
+                                                         self.boolvec_for_CG_ineqs.astype(int))
+                                               )])
+
+    @cached_property
+    def sparse_equalities_OLD(self) -> coo_matrix:
```

Comment:
@eliewolfe remove this?
```suggestion
    def sparse_equalities_OLD(self) -> coo_matrix:
```